plugin/validator_test.go

```diff
@@ -956,6 +956,889 @@ func TestValidator_SchemaAsJSON(t *testing.T) {
+		},
+		{
+			name: "valid auto_timestamp_ms",
+			config: `{
+				"auto_timestamp_ms": 1.234,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid auto_timestamp_ms",
+			config: `{
+				"auto_timestamp_ms": "1.234",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"auto_timestamp_ms":"expected a number"}}`,
+		},
+		{
+			name: "valid name",
+			config: `{
+				"name": "foo.-_~",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid name",
+			config: `{
+				"name": "foo.-_~!",
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{
+				"name":"invalid value 'foo.-_~!': it must only contain alphanumeric and '., -, _, ~' characters"
+				}}`,
+		},
+		{
+			name: "valid utf8_name",
+			config: `{
+				"utf8_name": "foo.-_~",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid name",
+			config: `{
+				"utf8_name": "foo.-_~!",
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{
+				"utf8_name":"invalid value 'foo.-_~!': the only accepted ascii characters are alphanumerics or ., -, _, and ~"
+				}}`,
+		},
+		{
+			name: "valid sni",
+			config: `{
+				"sni": "www.konghq.com",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid sni with port",
+			config: `{
+				"sni": "www.konghq.com:8080",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"sni":"must not have a port"}}`,
+		},
+		{
+			name: "invalid sni as IP",
+			config: `{
+				"sni": "192.0.2.0",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"sni":"must not be an IP"}}`,
+		},
+		{
+			name: "valid tags",
+			config: `{
+				"tags": [
+					"tag1"
+				],
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "valid tags",
+			config: `{
+				"tags": [
+					"tag1",
+					"tag2!"
+				],
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid tags with invalid character",
+			config: `{
+				"tags": [
+					"tag1",
+					"tag2,"
+				],
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{"tags":["[2] = invalid tag 'tag2,': ` +
+				`expected printable ascii (except ` + "`,` and `/`)" + ` or valid utf-8 sequences"]}}`,
+		},
+		{
+			name: "invalid tags with invalid character",
+			config: `{
+				"tags": [
+					"tag1",
+					"tag2/"
+				],
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{"tags":["[2] = invalid tag 'tag2/': ` +
+				`expected printable ascii (except ` + "`,` and `/`)" + ` or valid utf-8 sequences"]}}`,
+		},
+		{
+			name: "invalid tags with ASCII utf8",
+			config: `{
+				"tags": [
+					"tagõ"
+				],
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{"tags":["invalid tag 'tagõ': ` +
+				`expected printable ascii (except ` + "`,` and `/`)" + ` or valid utf-8 sequences"]}}`,
+		},
+		{
+			name: "valid sources with ip only",
+			config: `{
+				"sources": [
+					{
+						"ip": "192.0.2.1"
+					}
+				],
+				"tag": "test"
+			}`,
+		},
```

Comment:
IPv6 is needed.

Reply:
https://github.com/Kong/goks/pull/14/commits/c560ec1384ea55c6fb567a3e873000b1ea8c33df